opentelemetry-sdk/src/metrics/periodic_reader.rs

```diff
@@ -350,6 +344,8 @@ impl PeriodicReaderInner {
             return Ok(());
         }
 
+        otel_debug!(name: "PeriodicReaderMetricsCollected", count = rm.scope_metrics.len());
```

Comment:
This would return the count of meters, right? I think what we want is something like this:

```rust
// Count the total number of metrics collected.
let metrics_count = rm.scope_metrics.iter().fold(0, | count, scope_metrics | count + scope_metrics.metrics.len());
otel_debug!(name: "PeriodicReaderMetricsCollected", count = metrics_count);
```